### PR TITLE
Cls17 18 taking a vacation

### PR DESCRIPTION
**Pull request by 09041985**
@dmitrytokarev Take a look please, a can't write test function for final total sum.

### CODE REVIEW

File: tests/test_taling_a_vacation.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import sys
+
+import pytest
+
+from codecademy.taling_a_vacation import hotel_cost
+from codecademy.taling_a_vacation import plane_ride_cost
+from codecademy.taling_a_vacation import rental_car_cost
+from codecademy.taling_a_vacation import trip_cost
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: Thanks for correctly ordering and organizing your imports! Looks great!

File: tests/test_taling_a_vacation.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+@pytest.mark.parametrize("test_hotel_cost_input,test_hotel_cost_expected", [
+    (-sys.maxsize - 1, (-sys.maxsize - 1)*140),
+    (-sys.maxsize, -sys.maxsize*140),
+    (-1, -140),
+    (0, 0),
+    (1, 140),
+    (3037000499, 425180069860),
+    (sys.maxsize - 1, (sys.maxsize-1)*140),
+    (sys.maxsize, sys.maxsize*140)
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: Well here is a different story with the hotel.... It's a real life scenario, so the function uses number of days as input to count cost. By definition number of day can't be negative number - otherwise the hotel will have to pay the customer ))))
also think about what will be the maximum value for the upper boundary. It is not sys.maxsize because that is more then the age of the Universe 😄   https://en.wikipedia.org/wiki/Age_of_the_universe
So it should be some reasonable number - i think a person can stay for more than 1 year in the hotel.
So just google about longest hotel stay :)

File: tests/test_taling_a_vacation.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+])
+def test_hotel_cost(test_hotel_cost_input, test_hotel_cost_expected):
+    assert hotel_cost(test_hotel_cost_input) == test_hotel_cost_expected
+
+
+@pytest.mark.parametrize("test_plane_ride_cost_input,test_plane_ride_cost_expected", [
+    ("Charlotte", 183),
+    ("Tampa", 220),
+    ("Pittsburgh", 222),
+    ("Los Angeles", 475)
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: What about some negative tests? Some unexpected city name? Murmansk? LOL- here you have to expect exception:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

File: tests/test_taling_a_vacation.py (hunk 2 above)
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: Also you do need some of the negative inputs - but must expected some exception. Hint: you have to take care of the invalid inputs in the function - yes it's a bug there and you have to fix it :)

File: tests/test_taling_a_vacation.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    (5, 40*5-20),
+    (6, 40*6-20),
+    (7, 40*7-50),
+    (8, 40*8-50),
+    (3037000499, 3037000499*40-50),
+    (sys.maxsize-1, (sys.maxsize-1)*40-50),
+    (sys.maxsize, sys.maxsize*40-50)
+])
+def test_rental_car_cost(test_rental_car_cost_input, test_rental_car_cost_expected):
+    assert rental_car_cost(test_rental_car_cost_input) == test_rental_car_cost_expected
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: same story as with hotel - you can't rent for number of days that is older then the Universe ))
Well that is a good negative test - and you should expect some exception.
Also need to redo the equivalence partitions for this function (same as for the hotel stay cost)

File: tests/test_taling_a_vacation.py [hunk 5]
```diff
@@ -31,6 +35,13 @@ def test_plane_ride_cost(test_plane_ride_cost_input, test_plane_ride_cost_expect
     assert plane_ride_cost(test_plane_ride_cost_input) == test_plane_ride_cost_expected
 
 
+def test_plane_ride_cost_negative():
+    try:
+        plane_ride_cost('sochi')
+    except ValueError as kaka:
+        assert kaka.message == "We Don't fly to the city sochi"
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: i remember kaka 😆

File: tests/test_taling_a_vacation.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    (2, 2*140),
+    (7, 7*140),
+    (31, 31*140),
+    (365, 365*140)
+])
+def test_hotel_cost(test_hotel_cost_input, test_hotel_cost_expected):
+    assert hotel_cost(test_hotel_cost_input) == test_hotel_cost_expected
+
+
+def test_hotel_cost_negative():
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: test function name doesn't correspond to the actual test:
name suggests that you will call hotel_cost() with some negative number but you call hotel_cost() with a string.
Well we need both of these tests (negative number and different type of argument (string, boolean, etc.))

File: tests/test_taling_a_vacation.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    (0, 0),
+    (1, 40),
+    (2, 80),
+    (3, 40*3-20),
+    (4, 40*4-20),
+    (5, 40*5-20),
+    (6, 40*6-20),
+    (7, 40*7-50),
+    (8, 40*8-50),
+    (365, 40*365-50)
```
Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: these are all positive tests (all input values are positive integers), could you add some negative tests?
(different input type, negative numbers, etc.)

Comment (dmitrytokarev) [inline tests/test_taling_a_vacation.py]: hint: you will have to add a logic in the actual function to handle invalid inputs (like you did with hotel_cost())

**Comment by dmitrytokarev** [comment]
@09041985 please address my last 2 comments and this PR is good to go!

File: report.html [hunk 8]
```diff
@@ -0,0 +1,928 @@
+<!DOCTYPE html>
```
Comment (dmitrytokarev) [inline report.html]: The report file is supposed to be generated for each test run. These no need to commit it. Please remove it.

File: tests/test_taking_a_vacation.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    (365, 365*140)
+])
+def test_hotel_cost(test_hotel_cost_input, test_hotel_cost_expected):
+    assert hotel_cost(test_hotel_cost_input) == test_hotel_cost_expected
+
+
+@pytest.mark.parametrize("test_hotel_cost_negative_input", [1, -1, "blah", False, None])
+def test_hotel_cost_negative(test_hotel_cost_negative_input):
+    try:
+        hotel_cost(test_hotel_cost_negative_input)
```
Comment (dmitrytokarev) [inline tests/test_taking_a_vacation.py]: @09041985 i totally missed this, sorry
this test expects an exception, but if there is no exception thrown your test will do nothing about that.
basically if `hotel_cost()` returns normally you don't have a mechanism to fail the test.
This situation is called that your test may `falsely pass` which is a very bad practice!
Here if your call to `hotel_cost()` passes then Python interpreter will look for next line in the `try` block - you don't have any lines there so it will "jump out" of the try-except block and look for next line  - you don't have any >> test is over and since you didn't throw any exceptions test will pass.
To fix this you need to fail on the situation when `hotel_cost()` doesn't through exception - so just add a line `assert False, "Expected an exception from hotel_cost"` or after hotel_cost() call.
or better save result of calling hotel_cost and add that result to your custom message to `assert False` so it'll be easier to debug failed test.
Please apply this fix to all tests that expect exceptions.

File: report.html (hunk 8 above)
Comment (dmitrytokarev) [inline report.html]: @09041985 please see my previous comment ^^

**Comment by dmitrytokarev** [comment]
@09041985 please update this branch with master and merge

**Comment by dmitrytokarev** [comment]
:shipit: 
:shipit: 
:shipit:
